### PR TITLE
bazel: Update rules_rust

### PR DESCRIPTION
**Pull request by keith**
Some small things moved around since the last update

Closes https://github.com/envoyproxy/envoy/issues/20611

Signed-off-by: Keith Smiley <keithbsmiley@gmail.com>

### CODE REVIEW

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -1056,13 +1056,13 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_name = "Bazel rust rules",
         project_desc = "Bazel rust rules (used by Wasm)",
         project_url = "https://github.com/bazelbuild/rules_rust",
-        version = "82b650d5d0709ae4c0ee8584f4ed92112ba11d67",
-        sha256 = "d087851b76204935f7f23c172eb0d136c09720b8484d8151019523652ce77004",
+        version = "635da93206c2761d9db58ed82e194d2af053faf6",
+        sha256 = "aeaa5b98891c646c5292d36f1ce84682f353395aacddafd6c3d6e0db3026eb85",
```
Comment (phlax) [inline bazel/repository_locations.bzl:1060]: im wondering if we should use v0.1.0 which seems to have been released recently (https://github.com/bazelbuild/rules_rust/releases)

Comment (moderation) [reply bazel/repository_locations.bzl:1060]: +1 for `0.1.0` - https://github.com/bazelbuild/rules_rust/releases/tag/0.1.0

**Review by moderation — CHANGES_REQUESTED** [review]

File: bazel/repository_locations.bzl (hunk 1 above)
Comment (keith) [reply bazel/repository_locations.bzl]: Nice I hadn't seen that, thanks, updated

Comment (phlax) [reply bazel/repository_locations.bzl]: version `0.2.0` is now out

**Comment by keith** [comment]
Ok I bumped to 0.2.0 and hopefully got past the wasm issue here

**Review by moderation — APPROVED** [review]
/lgtm deps

**Pull request merged**